baseband/vdif/tests/test_vdif.py

```diff
@@ -897,6 +897,31 @@ def test_subset(self, tmpdir):
             check = fhn.read()
             assert np.all(check == data4x[(slice(None),) + subset_md])
 
+    def test_stream_verify(self, tmpdir):
+        testverifyfile = str(tmpdir.join('testverify.vdif'))
+        # Make a file with a sync pattern error in the second set of frames.
+        with vdif.open(SAMPLE_FILE, 'rb') as fh, \
+                vdif.open(testverifyfile, 'wb') as fw:
+            fr = fh.read_frameset()
+            fr.tofile(fw)
+            fr = fh.read_frameset()
+            fr.frames[2].header.mutable = True
+            fr.frames[2].header['sync_pattern'] = 0xabbaabba
+            fr.tofile(fw)
+
+        with vdif.open(SAMPLE_FILE, 'rs') as fh, \
+                vdif.open(testverifyfile, 'rs') as fn:
+            assert fn.verify is True
+            # This should fail at the second frameset.
+            with pytest.raises(AssertionError):
+                fn.read()
+            assert fn.tell() == 20000
+            fn.verify = False
+            assert fn.verify is False
+            # Seek fh to second frameset and check that we can properly read.
+            fh.seek(20000)
+            assert np.all(fn.read() == fh.read())
+
```

Comment:
Nice that you wrote a test! Can we also check that opening it with `verify=False` works?